src/form.tsx

```diff
@@ -86,7 +89,7 @@ async function fetchData({
 }: FetchDataOptions): Promise<Response> {
   // patch must be in all caps: https://github.com/github/fetch/issues/254
   const method = methodArg.toUpperCase();
-  const url = new URL(urlArg);
+  const url = new URL(urlArg, window.location.href);
 
```

Comment:
Should we only do this if urlArg is relative? What if someone provides the full url as action?

Reply:
If someone passes a full URL, it replaces the window location:

![image](https://user-images.githubusercontent.com/19603573/134524192-d5e747dd-d1a5-4eeb-811f-bcdddb441220.png)